src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,304 @@
+            }
+            fw.close();
+        } catch (IOException e) {
+            System.out.println("Error while saving to file amon_gus.txt.");
+        }
+    }
+    public static void deleteTask(String stringIndex, ArrayList<Task> tasks) {
+        int index;
+        try {
+            index = Integer.parseInt(stringIndex);
+        } catch (NumberFormatException e) {
+            System.out.println("Not a valid task number!");
+            return;
+        }
+        try {
+            tasks.remove(index - 1);
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("Task not in list!");
+        }
+    }
+
+    public static void handleT (ArrayList<Task> tasks, String[] parameters) throws DataFileCorruptedException {
+        try {
+            String description = parameters[2];
+            ToDo newTodo = new ToDo(description);
+            newTodo.setStatus(parameters[1].equals("✓"));
+            tasks.add(newTodo);
+        } catch (Exception e) {
+            throw new DataFileCorruptedException();
+        }
+    }
+
+    public static void handleD (ArrayList<Task> tasks, String[] parameters) throws DataFileCorruptedException{
+        try {
+            String description = parameters[2];
+            String deadline = parameters[3];
+            Deadline newDeadline = new Deadline(description, deadline);
+            newDeadline.setStatus(parameters[1].equals("✓"));
+            tasks.add(newDeadline);
+        } catch (Exception e) {
+            throw new DataFileCorruptedException();
+        }
+    }
+
+    public static void handleE (ArrayList<Task> tasks, String[] parameters) throws DataFileCorruptedException{
+        try {
+            String description = parameters[2];
+            String at = parameters[3];
+            Deadline newEvent = new Deadline(description, at);
+            newEvent.setStatus(parameters[1].equals("✓"));
+            tasks.add(newEvent);
+        } catch (Exception e) {
+            throw new DataFileCorruptedException();
+        }
+    }
+    public static void addToTasks(String input, ArrayList<Task> tasks) {
+        String[] parameters = input.split(" # ");
+        try {
+            switch (parameters[0]) {
+            case "T": {
+                handleT(tasks, parameters);
+                break;
+            }
+            case "D": {
+                handleD(tasks, parameters);
+                break;
+            }
+            case "E": {
+                handleE(tasks, parameters);
+                break;
+            }
+            default: {
+                System.out.println("Data file data/amon_gus.txt corrupted!");
+                break;
+            }
+            }
+        } catch (DataFileCorruptedException e) {
+            System.out.println("Data file data/amon_gus.txt corrupted!");
+        }
+
+    }
+
+
+    public static void readFile(ArrayList<Task> tasks) throws FileNotFoundException{
+        File f = new File(AMON_GUS_PATH);
```

Comment:
Please consider using the name "file" instead of "f" as the name of this variable.